lua/neo-tree/sources/common/components.lua

```diff
@@ -77,6 +77,9 @@ end
 M.diagnostics = function(config, node, state)
   local diag = state.diagnostics_lookup or {}
   local diag_state = diag[node:get_id()]
+  if node:is_expanded() then
```

Comment:
This should probably be:
```suggestion
  if node.type == "directory" and node:is_expanded() then
```

Technically there are nested files and diagnostic messages (in the external diagnostics source) that can be expanded but should still show the diagnostics markers.